### ISSUE TITLE
Pre-processing of face images/pretrained models/quantitative results

### ISSUE THREAD

**Issue by Werty18** [CGCL-codes/AMT-GAN#6]
Hi, thanks for the nice work. We have the following questions

1) - Are you doing any pre-processing while giving face image to FR model (like detecting face keypoints and crop it as done in facial recognition training works). If not, do you think that change in the background color/texture during makeup transfer can impact performance?

2) - There is only one pre-trained model available (mobileface, trained on ensemble of other three models). Do you have plans to upload pretrained models of facenet, IR152, and IRSE50?

3) - There are 9 reference makeup images. Can you please let us know against which image have you evaluated your quantitative results in Table-1 of paper? Similarly, Table-2 results are for which model (mobileface or some other)?

Regards

**Comment by SheltonLiu-N** [comment]
Thanks for your interest!
1. We have tried pre-processing the face images before feeding them into the FR models, including the training phase (for adversarial loss) and verification phase, however, we don't find an obvious difference between with and without pre-processing (for adversarial loss or verification). So in the evaluations, we don't involve the pre-processing. We witness that, AMT-GAN mainly changes pixels of the face, and the modification of the background may be caused by the nature of makeup transfer (either clean or adversarial makeup transfer GAN). So we don't think pre-processing can affect the performance, but it's still worth a try.
2. Sorry, since the framework of AMT-GAN is shared completely in this repository, we may not upload pretrained models of facenet, IR152, and IRSE50 recently. You can follow README and customize the training process, then you will get similar results that we described in our paper.
3. We evaluate the AMT-GAN on all the 10 reference makeup images and 4 target models, the quantitative results are computed by average.

**Comment by Werty18** [comment]
Thanks for the detailed response. 

**1.**  We found that there are 9 reference images in the assets folder. Can you please upload the 10th image so that we can reproduce results. 

**2.**  Further, there is only 1 target image provided in the asset folder (in paper there are 4). Can you kindly upload the other three? 

**3.**  As you have mentioned, quantitative results are averaged for 10 ref images and 4 target models. Have you also averaged results over 4 target identities or only for the one target identity provided in the asset folder (in assets-->datasets-->target  folder)?

**4.** Have you also use ensemble approach for PGD and other baseline methods in Table-1?

Thanks and regrads.

**Comment by SheltonLiu-N** [comment]
1. Sorry! There is 1 reference image missing for an unknown reason, here is the all reference images
[reference.zip](https://github.com/CGCL-codes/AMT-GAN/files/9078210/reference.zip)
2. Here is the 4 target/test pairs:
[pair.zip](https://github.com/CGCL-codes/AMT-GAN/files/9078233/pair.zip)
3. Yes.
4. No, one of the competitor methods (TIP-IM) is hard to apply to the ensemble scheme, because it builds an embedding map for target identity adaptive change.
